staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -693,6 +724,29 @@ type PersistentVolumeClaimStatus struct {
 	// +mapType=granular
 	// +optional
 	AllocatedResourceStatuses map[ResourceName]ClaimResourceStatus `json:"allocatedResourceStatuses,omitempty" protobuf:"bytes,7,rep,name=allocatedResourceStatuses"`
+	// volumeAttributesClassName is the name of VolumeAttributesClass which is latest applied by
+	// the CSI driver. This field is not set by default but when modification is complete,
+	// it's set to the applied VolumeAttributesClass.
+	// This is an alpha field and requires enabling VolumeAttributesClass feature.
+	// +featureGate=VolumeAttributesClass
+	// +optional
+	VolumeAttributesClassName *string `json:"volumeAttributesClassName,omitempty" protobuf:"bytes,8,opt,name=volumeAttributesClassName"`
+	// VolumeAttributesModifyStatus stores status of modification operation. It can be in any of following states:
+	//  - PersistentVolumeClaimControllerModifyVolumeInProgress
+	//    State set when modify volume controller starts modifying the volume in control-plane.
+	//  - PersistentVolumeClaimControllerModifyVolumeFailed
+	//    State set when modify volume has failed in modify volume controller with a terminal error.
+	//    Transient errors such as timeout should not set this status and should leave VolumeAttributesModifyStatus
+	//    unmodified, so as modify volume controller can resume the volume modification.
+	// For example: if modifying a PVC's volume attributes, this field can be one of the following states:
+	//   - pvc.status.volumeAttributesModifyStatus = "ControllerModifyVolumeInProgress"
+	//   - pvc.status.volumeAttributesModifyStatus = "ControllerModifyVolumeFailed"
+	// When this field is not set, it means that no volume attributes modification operation is in progress for the given PVC.
```

Comment:
So both an unset field and the "PersistentVolumeClaimControllerModifyVolumeInProgress" enum value mean that something is in-progress? What's the difference?

Reply:
A `nil` value means that no operation is in progress.